cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -1279,3 +1280,67 @@ def nfs_storage(mount):
         return
 
     set_state('nfs.configured')
+
+
+@when('endpoint.docker-registry.ready')
+@when_not('kubernetes-worker.registry.configured')
+def configure_registry():
+    '''Add docker registry config when present.'''
+    registry = endpoint_from_flag('endpoint.docker-registry.ready')
+    netloc = registry.registry_netloc
+
+    # handle tls data
+    cert_subdir = netloc
+    insecure_opt = {'insecure-registry': netloc}
+    if registry.has_tls():
+        # ensure the CA that signed our registry cert is trusted
+        install_ca_cert(registry.tls_ca, name='juju-docker-registry')
+        # remove potential insecure docker opts related to this registry
+        manage_docker_opts(insecure_opt, remove=True)
+        manage_registry_certs(cert_subdir, remove=False)
+    else:
+        manage_docker_opts(insecure_opt, remove=False)
+        manage_registry_certs(cert_subdir, remove=True)
+
+    # handle auth data
+    if registry.has_auth_basic():
+        hookenv.log('Logging into docker registry: {}.'.format(netloc))
+        cmd = ['docker', 'login', netloc,
+               '-u', registry.basic_user, '-p', registry.basic_password]
+        subprocess.check_call(cmd)
+    else:
+        hookenv.log('Disabling auth for docker registry: {}.'.format(netloc))
+        # NB: it's safe to logout of a registry that was never logged in
+        subprocess.check_call(['docker', 'logout', netloc])
```

Comment:
I assume logging out twice or logging in twice isn't a big deal to docker. Is that correct?

Reply:
@hyperbolic2346 you are correct; multiple logins/logouts are no big deal:

```bash
ubuntu@ip-172-31-5-130:~$ docker login 172.31.24.194:5000 -u admin -p admin
Login Succeeded

ubuntu@ip-172-31-5-130:~$ docker login 172.31.24.194:5000 -u admin -p admin
Login Succeeded

ubuntu@ip-172-31-5-130:~$ docker login 172.31.24.194:5000 -u admin -p admin
Login Succeeded

ubuntu@ip-172-31-5-130:~$ docker logout 172.31.24.194:5000
Removing login credentials for 172.31.24.194:5000

ubuntu@ip-172-31-5-130:~$ docker logout 172.31.24.194:5000
Not logged in to 172.31.24.194:5000

ubuntu@ip-172-31-5-130:~$ docker logout 172.31.24.194:5000
Not logged in to 172.31.24.194:5000

ubuntu@ip-172-31-5-130:~$ echo $?
0
```